[mariadb] Change default binary log format from MIXED to ROW

s10 opened the pull request:
mariadb: Change default binary log format from MIXED to ROW
This will increase the size of binary logs but will make them more consistent and usable for mariadb->galera migration when AUTO_INCREMENT is being used.
Note: the overall size of binary logs and binary log backups will increase after this change

galkindmitrii reviewed (dismissed):
lgtm, but I think @IvoGoman should have a look

IvoGoman commented:
This should be fine from `go-maria-sync`-side, as long as this is `RowsEventV1`. I was not yet able to start working on the `RowsEventV2` support.

I faintly remember some issues with enabling [binlog_annotate_row_events](https://mariadb.com/kb/en/replication-and-binary-log-system-variables/#binlog_annotate_row_events) but since this seems to be already enabled by default for quite some time there should be no issue. 

In case it's not the v2 just let me know once you start rolling this out in QA then I could double-check if the replication still works :)

s10 committed:
[mariadb] Change default binary log format from MIXED to ROW

Change the default binary log format from MIXED to ROW.

This would make possible to use galera clustered MariaDB or MySQL
as a target for backup restore.
